examples/03_connectivity/plot_group_level_connectivity.py

```diff
@@ -0,0 +1,259 @@
+from nilearn import plotting
+
+for n, matrix in enumerate(correlation_matrices[:3]):
+    plotting.plot_connectome(matrix, msdl_coords, display_mode='x',
+                             title='correlation, subject {0}'.format(n))
+
+plot_matrix(mean_correlation_matrix, 'thresholded mean correlation')
+
+###############################################################################
+# Look at blocks structure, reflecting functional networks.
+
+###############################################################################
+# Examine partial correlations
+# ----------------------------
+# We can also study **direct connections**, revealed by partial correlation
+# coefficients. We just change the `ConnectivityMeasure` kind
+partial_correlation_measure = ConnectivityMeasure(kind='partial correlation')
+
+###############################################################################
+# and repeat the previous operation.
+partial_correlation_matrices = partial_correlation_measure.fit_transform(
+    adhd_subjects)
+
+###############################################################################
+# Most of direct connections are weaker than total connections: The mean matrix
+# has more marked blocks, visible even without thresholding.
+for n, matrix in enumerate(partial_correlation_matrices[:3]):
+    plotting.plot_connectome(matrix, msdl_coords, display_mode='x',
+                             title='partial, subject {0}'.format(n))
+
+plot_matrix(partial_correlation_measure.mean_, 'mean partial correlation')
+
+###############################################################################
+# Extract robust group connectivity
+# ---------------------------------
+
+###############################################################################
+# We can use **both correlations and partial correlations** to capture
+# reproducible connectivity patterns at the group-level and build a **robust**
+# ** group connectivity matrix**. This is done by the **tangent** kind.
+tangent_measure = ConnectivityMeasure(kind='tangent')
+
+###############################################################################
+# We fit our ADHD group and get the group connectivity matrix stored as
+# in `mean_`, and individual deviation matrices of each subject from it.
+tangent_matrices = tangent_measure.fit_transform(adhd_subjects)
+
+tangent_vectors = sym_matrix_to_vec(tangent_matrices, discard_diagonal=True)
+chisquare_values, p_values = stats.chisquare(tangent_vectors.T.dot(tangent_vectors),
+                                             ddof=correlation_vectors.shape[1],
+axis=0)
+p_values = vec_to_sym_matrix(p_values, diagonal=np.ones(n_regions))
+
+###############################################################################
+# `tangent_matrices` are complex features that do not reflect brain connections
+# directly. An additionnal step is needed to recover connectivity.
+connectivity_matrices = tangent_measure.inverse_transform(tangent_matrices)
```

Comment:
Yes. In all cases `inverse_transform` gives individual connectivity matrices, whether or not vectorization happened. So if it was not asked for vectorization, `inverse_transform` does nothing for `correlation` and `partial correlation`. However for `tangent`, what I am calling `tangent_matrices` are not true connectivity matrices because for instance they can have negative values on the diagonal. The only way to recover the `connectivity_matrices` is to inverse transform.
I can't figure out a way to be more pedagogic on this mess, so if you have suggestions...